lvmd/embed.go

```diff
@@ -0,0 +1,128 @@
+package lvmd
+
+import (
+	"context"
+
+	"github.com/topolvm/topolvm/lvmd/proto"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	gproto "google.golang.org/protobuf/proto"
+)
+
+// NewEmbeddedServiceClients creates clients locally calling instead of using gRPC.
+func NewEmbeddedServiceClients(ctx context.Context, dcmapper *DeviceClassManager, ocmapper *LvcreateOptionClassManager) (
+	proto.LVServiceClient,
+	proto.VGServiceClient,
+) {
+	vgServiceServerInstance, notifier := NewVGService(dcmapper)
+	lvServiceServerInstance := NewLVService(dcmapper, ocmapper, notifier)
+	caller := &localCaller{
+		lvServiceServer: lvServiceServerInstance,
+		vgServiceServer: vgServiceServerInstance.(*vgService),
+	}
+	caller.vgWatch = &localWatch{ctx: ctx, watch: make(chan any)}
+	go caller.vgServiceServer.Watch(&proto.Empty{}, caller.vgWatch)
+	return caller, caller
+}
+
+// localCaller is a struct holding indirections to the local lvmd server.
+// It implements both the LVServiceClient and VGServiceClient interfaces.
+// It also includes a watch for redirecting the vg watch to the local caller.
+type localCaller struct {
```

Comment:
I prefer to rename this struct's name to something like `embeddedServiceClient`.